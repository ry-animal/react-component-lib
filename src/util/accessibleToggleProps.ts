export function accessibleToggleProps(
  label: string,
  controls: string,
  expanded: boolean,
  popupType?: "menu"
) {
  return {
    "aria-controls": controls,
    "aria-expanded": expanded,
    "aria-haspopup": popupType || true,
    "aria-label": label,
    "aria-pressed": expanded,
    role: "button",
  };
}
