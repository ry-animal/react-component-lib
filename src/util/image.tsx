import { ISelectedFile } from "src/components/Forms/FileUpload/ISelectedFile";

export const makeDataUrl = (selectedFile: ISelectedFile) =>
  URL.createObjectURL(selectedFile.file);
