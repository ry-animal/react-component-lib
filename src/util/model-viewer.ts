/*
 * This file is a hacky way of allowing any other component to
 * use the <model-viewer /> tag just by doing:
 *   import "../../../util/model-viewer";
 *
 * The authors of the package do not expose it in a React-friendly
 * way, see: https://github.com/google/model-viewer/issues/1502
 *
 * The component documentation is at https://modelviewer.dev/docs/
 */

import "@google/model-viewer";

declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace JSX {
    export interface IntrinsicElements {
      "model-viewer": {
        src?: string;
        alt?: string;
        poster?: string;
        autoplay?: boolean;
        exposure?: number;
        "camera-controls"?: boolean;
        "environment-image"?: string;
        "seamless-poster"?: string;
        loading?: "auto" | "lazy" | "eager";
        reveal?: "auto" | "interaction" | "manual";
      } & React.DetailedHTMLProps<
        React.HTMLAttributes<HTMLElement>,
        HTMLElement
      >;
    }
  }
}
