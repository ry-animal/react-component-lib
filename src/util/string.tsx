const DEFAULT_LOCALE = "en-US";

export const truncateEthAddress = (str: string): string => {
  if (str.length) {
    const numStartingCharacters = 6;
    const numTrailingCharacters = 4;
    return `${str.substring(0, numStartingCharacters)}...${str.substring(
      str.length - numTrailingCharacters,
      str.length
    )}`;
  }
  return "";
};

export const formatUSDPriceString = (value: number): string => {
  const minimumFractionDigits = 2;
  const maximumFractionDigits = 2;
  return `$${value.toLocaleString(DEFAULT_LOCALE, {
    minimumFractionDigits,
    maximumFractionDigits,
  })}`;
};

export const formatETHPriceString = (priceInEth: number): string => {
  const minimumFractionDigits = 0;
  const maximumFractionDigits = 8;
  const maxDigitsWithDecimal = 8;

  let formatted = `${priceInEth
    .toLocaleString(DEFAULT_LOCALE, {
      minimumFractionDigits,
      maximumFractionDigits,
    })
    .replace(/,/gu, "")
    .substring(0, maxDigitsWithDecimal)}`;

  if (formatted.endsWith(".")) {
    formatted = formatted.slice(0, -1);
  }
  return formatted;
};

const evaluateNumberLessThanOne = (num: number) => {
  if (num === 0) {
    return "0";
  }

  const minDecimalValue = 0.0001;
  if (num < minDecimalValue) {
    return minDecimalValue.toString();
  }
  // To round decimal to 4 digits
  const maxDigits = 10000;
  return `${Math.round(num * maxDigits) / maxDigits}`;
};

// Abbreviate Number https://stackoverflow.com/a/9462382

export const abbreviateNumber = (num: number): string => {
  const lookup = [
    { value: 1, symbol: "" },
    { value: 1e3, symbol: "k" },
    { value: 1e6, symbol: "M" },
    { value: 1e9, symbol: "B" },
    { value: 1e12, symbol: "T" },
    { value: 1e15, symbol: "P" },
    { value: 1e18, symbol: "E" },
  ];

  const rx = /\.0+$|(?<id>\.[0-9]*[1-9])0+$/u;
  const digits = 1;

  const item = lookup
    .slice()
    .reverse()
    .find((x) => num >= x.value);
  return item
    ? (num / item.value).toFixed(digits).replace(rx, "$1") + item.symbol
    : evaluateNumberLessThanOne(num);
};

export const getDomainFromRegex = (url: string) => {
  const matches = url.match(/:\/\/(?<id>.[^/]+)/u);
  // Extract domain (will be null if no match is found)
  return matches && matches[1];
};
