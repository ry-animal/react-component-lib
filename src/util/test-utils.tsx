import { render, RenderOptions } from "@testing-library/react";
import "jest-styled-components";
import * as React from "react";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { LIGHT_THEME } from "../theme";

interface ProvidersProps {
  children: React.ReactNode;
}

function Providers({ children }: ProvidersProps) {
  return (
    <BrowserRouter>
      <ThemeProvider theme={LIGHT_THEME}>{children}</ThemeProvider>
    </BrowserRouter>
  );
}

const customRender = (
  ui: JSX.Element,
  options?:
    | RenderOptions<
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement
      >
    | undefined
) => render(ui, { wrapper: Providers, ...options });

// Duplicated instead of import "../../__mocks__/matchMediaMock"; because jest has trouble resolving imports
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

// Re-export everything
export * from "@testing-library/react";
// Override render method
export { customRender as render };
