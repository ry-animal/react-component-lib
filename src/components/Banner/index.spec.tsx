import { Banner, BannerProps } from ".";
import { render } from "../../util/test-utils";

const createPropsWithActions = (props: BannerProps) => ({
  ...props,
  dismissAction: () => {},
  primaryAction: {
    label: "Add Funds",
    onClick: () => {},
  },
  secondaryAction: {
    label: "Do it later",
    onClick: () => {},
  },
});

describe("<Banner/>", () => {
  const baseProps = {
    title: "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
    description:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et.",
    showDismissButton: false,
  };
  test("Default rendering matches snapshot", () => {
    expect(render(<Banner {...baseProps} />).asFragment()).toMatchSnapshot();
  });

  test("Default rendering matches snapshot with Actions", () => {
    expect(
      render(<Banner {...createPropsWithActions(baseProps)} />).asFragment()
    ).toMatchSnapshot();
  });

  test("Rendering matches snapshot with Dismiss Button enabled", () => {
    expect(
      render(
        <Banner
          {...createPropsWithActions({ ...baseProps, showDismissButton: true })}
        />
      ).asFragment()
    ).toMatchSnapshot();
  });

  test("Rendering matches snapshot with description", () => {
    expect(
      render(
        <Banner
          {...createPropsWithActions({
            ...baseProps,
            description: <p>hello world</p>,
          })}
        />
      ).asFragment()
    ).toMatchSnapshot();
  });
});
