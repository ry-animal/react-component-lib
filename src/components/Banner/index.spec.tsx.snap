// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner/> Default rendering matches snapshot 1`] = `
<DocumentFragment>
  .c5 {
  color: #151515;
  margin: 0;
  font-family: 'Poppins',sans-serif;
  font-weight: 600;
  font-size: 18px;
  line-height: 20px;
}

.c7 {
  color: #151515;
  margin: 0;
  font-family: 'Poppins',sans-serif;
  font-weight: 400;
  font-size: 18px;
  line-height: 28px;
}

.c0 {
  width: 100%;
  position: fixed;
  bottom: 64px;
}

.c1 {
  margin: auto;
  background-color: #FFFFFF;
  box-shadow: 0 2px 24px 0 rgba(0,0,0,0.2);
  border-radius: 8px;
  width: 75%;
  max-width: 1280px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  padding: 0 24px;
  height: 100%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c3 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  margin: 0;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 12px 24px;
}

@media (max-width:767px) {
  .c5 {
    font-size: 16px;
    line-height: 20px;
    -webkit-letter-spacing: 0em;
    -moz-letter-spacing: 0em;
    -ms-letter-spacing: 0em;
    letter-spacing: 0em;
  }
}

@media (max-width:767px) {
  .c7 {
    font-size: 16px;
    line-height: 24px;
  }
}

@media (max-width:768px) {
  .c1 {
    width: 100%;
  }
}

@media (max-width:768px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    padding-top: 24px;
  }
}

@media (max-width:1024px) {
  .c4 {
    display: none;
  }
}

@media (max-width:1024px) {
  .c3 {
    padding: 0 24px;
  }
}

@media (max-width:768px) {
  .c3 {
    text-align: center;
  }
}

@media (max-width:768px) {
  .c6 {
    margin: 8px;
  }
}

@media (max-width:768px) {
  .c8 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            class="c4"
          >
            <svg
              fill="none"
              height="32"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M25.2915 0.000269943C25.0883 0.00547672 24.8748 0.0315204 24.6665 0.0836031L2.99997 5.79189C1.24999 6.25022 0 7.85437 0 9.66685V28C0 30.1927 1.80728 32 3.99997 32H27.9998C30.1925 32 31.9997 30.1927 31.9997 28V10.6668C31.9997 8.47416 30.1925 6.66688 27.9998 6.66688H10.1666L25.3331 2.66691V5.33356H27.9998V2.66691C27.9998 1.16693 26.7237 -0.0205621 25.2915 0.000269943H25.2915ZM3.99997 9.33352H27.9998C28.755 9.33352 29.3331 9.91165 29.3331 10.6668V28C29.3331 28.7552 28.755 29.3334 27.9998 29.3334H3.99997C3.24477 29.3334 2.66664 28.7552 2.66664 28V10.6668C2.66664 9.91165 3.24477 9.33352 3.99997 9.33352ZM24.6665 17.3335C23.5623 17.3335 22.6665 18.2293 22.6665 19.3334C22.6665 20.4376 23.5623 21.3334 24.6665 21.3334C25.7706 21.3334 26.6664 20.4376 26.6664 19.3334C26.6664 18.2293 25.7706 17.3335 24.6665 17.3335Z"
                fill="#151515"
              />
            </svg>
          </div>
          <div>
            <p
              class="c5"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            </p>
            <div
              class="c6"
            >
              <p
                class="c7"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et.
              </p>
            </div>
          </div>
        </div>
        <div
          class="c8"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Banner/> Default rendering matches snapshot with Actions 1`] = `
<DocumentFragment>
  .c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex: unset;
  -ms-flex: unset;
  flex: unset;
  line-height: 1;
  gap: 8px;
}

.c10 {
  text-align: center;
  border: none;
  outline: none;
  overflow: hidden;
  cursor: pointer;
  pointer-events: auto;
  -webkit-transition: background-color 0.25s ease-out;
  transition: background-color 0.25s ease-out;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-height: 40px;
  font-weight: 600;
  padding: 0 28px;
  font-size: 16px;
  border-radius: 4px;
  background-color: #000000;
  color: #EEEEEE;
}

.c10:disabled {
  opacity: 20%;
  cursor: default;
  pointer-events: none;
}

.c10:hover:not([disabled]),
.c10:focus-visible:not([disabled]) {
  background-color: #5F5F5F;
}

.c12 {
  text-align: center;
  border: none;
  outline: none;
  overflow: hidden;
  cursor: pointer;
  pointer-events: auto;
  -webkit-transition: background-color 0.25s ease-out;
  transition: background-color 0.25s ease-out;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-height: 40px;
  font-weight: 600;
  padding: 0 28px;
  font-size: 16px;
  border-radius: 4px;
  background-color: transparent;
  box-shadow: inset 0 0 0 1px #000000;
  color: #000000;
}

.c12:disabled {
  opacity: 20%;
  cursor: default;
  pointer-events: none;
}

.c12:hover:not([disabled]),
.c12:focus-visible:not([disabled]) {
  box-shadow: inset 0 0 0 2px #000000;
}

.c5 {
  color: #151515;
  margin: 0;
  font-family: 'Poppins',sans-serif;
  font-weight: 600;
  font-size: 18px;
  line-height: 20px;
}

.c7 {
  color: #151515;
  margin: 0;
  font-family: 'Poppins',sans-serif;
  font-weight: 400;
  font-size: 18px;
  line-height: 28px;
}

.c0 {
  width: 100%;
  position: fixed;
  bottom: 64px;
}

.c1 {
  margin: auto;
  background-color: #FFFFFF;
  box-shadow: 0 2px 24px 0 rgba(0,0,0,0.2);
  border-radius: 8px;
  width: 75%;
  max-width: 1280px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  padding: 0 24px;
  height: 100%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c3 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  margin: 0;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 12px 24px;
}

.c9 {
  padding: 12px 8px;
}

@media (max-width:767px) {
  .c5 {
    font-size: 16px;
    line-height: 20px;
    -webkit-letter-spacing: 0em;
    -moz-letter-spacing: 0em;
    -ms-letter-spacing: 0em;
    letter-spacing: 0em;
  }
}

@media (max-width:767px) {
  .c7 {
    font-size: 16px;
    line-height: 24px;
  }
}

@media (max-width:768px) {
  .c1 {
    width: 100%;
  }
}

@media (max-width:768px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    padding-top: 24px;
  }
}

@media (max-width:1024px) {
  .c4 {
    display: none;
  }
}

@media (max-width:1024px) {
  .c3 {
    padding: 0 24px;
  }
}

@media (max-width:768px) {
  .c3 {
    text-align: center;
  }
}

@media (max-width:768px) {
  .c6 {
    margin: 8px;
  }
}

@media (max-width:768px) {
  .c8 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:768px) {
  .c9 button {
    width: 100%;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            class="c4"
          >
            <svg
              fill="none"
              height="32"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M25.2915 0.000269943C25.0883 0.00547672 24.8748 0.0315204 24.6665 0.0836031L2.99997 5.79189C1.24999 6.25022 0 7.85437 0 9.66685V28C0 30.1927 1.80728 32 3.99997 32H27.9998C30.1925 32 31.9997 30.1927 31.9997 28V10.6668C31.9997 8.47416 30.1925 6.66688 27.9998 6.66688H10.1666L25.3331 2.66691V5.33356H27.9998V2.66691C27.9998 1.16693 26.7237 -0.0205621 25.2915 0.000269943H25.2915ZM3.99997 9.33352H27.9998C28.755 9.33352 29.3331 9.91165 29.3331 10.6668V28C29.3331 28.7552 28.755 29.3334 27.9998 29.3334H3.99997C3.24477 29.3334 2.66664 28.7552 2.66664 28V10.6668C2.66664 9.91165 3.24477 9.33352 3.99997 9.33352ZM24.6665 17.3335C23.5623 17.3335 22.6665 18.2293 22.6665 19.3334C22.6665 20.4376 23.5623 21.3334 24.6665 21.3334C25.7706 21.3334 26.6664 20.4376 26.6664 19.3334C26.6664 18.2293 25.7706 17.3335 24.6665 17.3335Z"
                fill="#151515"
              />
            </svg>
          </div>
          <div>
            <p
              class="c5"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            </p>
            <div
              class="c6"
            >
              <p
                class="c7"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et.
              </p>
            </div>
          </div>
        </div>
        <div
          class="c8"
        >
          <div
            class="c9"
          >
            <button
              aria-label="Add Funds"
              class="c10"
            >
              <span
                class="c11"
              >
                Add Funds
              </span>
            </button>
          </div>
          <div
            class="c9"
          >
            <button
              aria-label="Do it later"
              class="c12"
            >
              <span
                class="c11"
              >
                Do it later
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Banner/> Rendering matches snapshot with Dismiss Button enabled 1`] = `
<DocumentFragment>
  .c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex: unset;
  -ms-flex: unset;
  flex: unset;
  line-height: 1;
  gap: 8px;
}

.c10 {
  text-align: center;
  border: none;
  outline: none;
  overflow: hidden;
  cursor: pointer;
  pointer-events: auto;
  -webkit-transition: background-color 0.25s ease-out;
  transition: background-color 0.25s ease-out;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-height: 40px;
  font-weight: 600;
  padding: 0 28px;
  font-size: 16px;
  border-radius: 4px;
  background-color: #000000;
  color: #EEEEEE;
}

.c10:disabled {
  opacity: 20%;
  cursor: default;
  pointer-events: none;
}

.c10:hover:not([disabled]),
.c10:focus-visible:not([disabled]) {
  background-color: #5F5F5F;
}

.c12 {
  text-align: center;
  border: none;
  outline: none;
  overflow: hidden;
  cursor: pointer;
  pointer-events: auto;
  -webkit-transition: background-color 0.25s ease-out;
  transition: background-color 0.25s ease-out;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-height: 40px;
  font-weight: 600;
  padding: 0 28px;
  font-size: 16px;
  border-radius: 4px;
  background-color: transparent;
  box-shadow: inset 0 0 0 1px #000000;
  color: #000000;
}

.c12:disabled {
  opacity: 20%;
  cursor: default;
  pointer-events: none;
}

.c12:hover:not([disabled]),
.c12:focus-visible:not([disabled]) {
  box-shadow: inset 0 0 0 2px #000000;
}

.c5 {
  color: #151515;
  margin: 0;
  font-family: 'Poppins',sans-serif;
  font-weight: 600;
  font-size: 18px;
  line-height: 20px;
}

.c7 {
  color: #151515;
  margin: 0;
  font-family: 'Poppins',sans-serif;
  font-weight: 400;
  font-size: 18px;
  line-height: 28px;
}

.c0 {
  width: 100%;
  position: fixed;
  bottom: 64px;
}

.c1 {
  margin: auto;
  background-color: #FFFFFF;
  box-shadow: 0 2px 24px 0 rgba(0,0,0,0.2);
  border-radius: 8px;
  width: 75%;
  max-width: 1280px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  padding: 0 24px;
  height: 100%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c3 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  margin: 0;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 12px 24px;
}

.c9 {
  padding: 12px 8px;
}

@media (max-width:767px) {
  .c5 {
    font-size: 16px;
    line-height: 20px;
    -webkit-letter-spacing: 0em;
    -moz-letter-spacing: 0em;
    -ms-letter-spacing: 0em;
    letter-spacing: 0em;
  }
}

@media (max-width:767px) {
  .c7 {
    font-size: 16px;
    line-height: 24px;
  }
}

@media (max-width:768px) {
  .c1 {
    width: 100%;
  }
}

@media (max-width:768px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    padding-top: 24px;
  }
}

@media (max-width:1024px) {
  .c4 {
    display: none;
  }
}

@media (max-width:1024px) {
  .c3 {
    padding: 0 24px;
  }
}

@media (max-width:768px) {
  .c3 {
    text-align: center;
  }
}

@media (max-width:768px) {
  .c6 {
    margin: 8px;
  }
}

@media (max-width:768px) {
  .c8 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:768px) {
  .c9 button {
    width: 100%;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            class="c4"
          >
            <svg
              fill="none"
              height="32"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M25.2915 0.000269943C25.0883 0.00547672 24.8748 0.0315204 24.6665 0.0836031L2.99997 5.79189C1.24999 6.25022 0 7.85437 0 9.66685V28C0 30.1927 1.80728 32 3.99997 32H27.9998C30.1925 32 31.9997 30.1927 31.9997 28V10.6668C31.9997 8.47416 30.1925 6.66688 27.9998 6.66688H10.1666L25.3331 2.66691V5.33356H27.9998V2.66691C27.9998 1.16693 26.7237 -0.0205621 25.2915 0.000269943H25.2915ZM3.99997 9.33352H27.9998C28.755 9.33352 29.3331 9.91165 29.3331 10.6668V28C29.3331 28.7552 28.755 29.3334 27.9998 29.3334H3.99997C3.24477 29.3334 2.66664 28.7552 2.66664 28V10.6668C2.66664 9.91165 3.24477 9.33352 3.99997 9.33352ZM24.6665 17.3335C23.5623 17.3335 22.6665 18.2293 22.6665 19.3334C22.6665 20.4376 23.5623 21.3334 24.6665 21.3334C25.7706 21.3334 26.6664 20.4376 26.6664 19.3334C26.6664 18.2293 25.7706 17.3335 24.6665 17.3335Z"
                fill="#151515"
              />
            </svg>
          </div>
          <div>
            <p
              class="c5"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            </p>
            <div
              class="c6"
            >
              <p
                class="c7"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et.
              </p>
            </div>
          </div>
        </div>
        <div
          class="c8"
        >
          <div
            class="c9"
          >
            <button
              aria-label="Add Funds"
              class="c10"
            >
              <span
                class="c11"
              >
                Add Funds
              </span>
            </button>
          </div>
          <div
            class="c9"
          >
            <button
              aria-label="Do it later"
              class="c12"
            >
              <span
                class="c11"
              >
                Do it later
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Banner/> Rendering matches snapshot with description 1`] = `
<DocumentFragment>
  .c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex: unset;
  -ms-flex: unset;
  flex: unset;
  line-height: 1;
  gap: 8px;
}

.c9 {
  text-align: center;
  border: none;
  outline: none;
  overflow: hidden;
  cursor: pointer;
  pointer-events: auto;
  -webkit-transition: background-color 0.25s ease-out;
  transition: background-color 0.25s ease-out;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-height: 40px;
  font-weight: 600;
  padding: 0 28px;
  font-size: 16px;
  border-radius: 4px;
  background-color: #000000;
  color: #EEEEEE;
}

.c9:disabled {
  opacity: 20%;
  cursor: default;
  pointer-events: none;
}

.c9:hover:not([disabled]),
.c9:focus-visible:not([disabled]) {
  background-color: #5F5F5F;
}

.c11 {
  text-align: center;
  border: none;
  outline: none;
  overflow: hidden;
  cursor: pointer;
  pointer-events: auto;
  -webkit-transition: background-color 0.25s ease-out;
  transition: background-color 0.25s ease-out;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-height: 40px;
  font-weight: 600;
  padding: 0 28px;
  font-size: 16px;
  border-radius: 4px;
  background-color: transparent;
  box-shadow: inset 0 0 0 1px #000000;
  color: #000000;
}

.c11:disabled {
  opacity: 20%;
  cursor: default;
  pointer-events: none;
}

.c11:hover:not([disabled]),
.c11:focus-visible:not([disabled]) {
  box-shadow: inset 0 0 0 2px #000000;
}

.c5 {
  color: #151515;
  margin: 0;
  font-family: 'Poppins',sans-serif;
  font-weight: 600;
  font-size: 18px;
  line-height: 20px;
}

.c0 {
  width: 100%;
  position: fixed;
  bottom: 64px;
}

.c1 {
  margin: auto;
  background-color: #FFFFFF;
  box-shadow: 0 2px 24px 0 rgba(0,0,0,0.2);
  border-radius: 8px;
  width: 75%;
  max-width: 1280px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  padding: 0 24px;
  height: 100%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c3 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  margin: 0;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 12px 24px;
}

.c8 {
  padding: 12px 8px;
}

@media (max-width:767px) {
  .c5 {
    font-size: 16px;
    line-height: 20px;
    -webkit-letter-spacing: 0em;
    -moz-letter-spacing: 0em;
    -ms-letter-spacing: 0em;
    letter-spacing: 0em;
  }
}

@media (max-width:768px) {
  .c1 {
    width: 100%;
  }
}

@media (max-width:768px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    padding-top: 24px;
  }
}

@media (max-width:1024px) {
  .c4 {
    display: none;
  }
}

@media (max-width:1024px) {
  .c3 {
    padding: 0 24px;
  }
}

@media (max-width:768px) {
  .c3 {
    text-align: center;
  }
}

@media (max-width:768px) {
  .c6 {
    margin: 8px;
  }
}

@media (max-width:768px) {
  .c7 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:768px) {
  .c8 button {
    width: 100%;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            class="c4"
          >
            <svg
              fill="none"
              height="32"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M25.2915 0.000269943C25.0883 0.00547672 24.8748 0.0315204 24.6665 0.0836031L2.99997 5.79189C1.24999 6.25022 0 7.85437 0 9.66685V28C0 30.1927 1.80728 32 3.99997 32H27.9998C30.1925 32 31.9997 30.1927 31.9997 28V10.6668C31.9997 8.47416 30.1925 6.66688 27.9998 6.66688H10.1666L25.3331 2.66691V5.33356H27.9998V2.66691C27.9998 1.16693 26.7237 -0.0205621 25.2915 0.000269943H25.2915ZM3.99997 9.33352H27.9998C28.755 9.33352 29.3331 9.91165 29.3331 10.6668V28C29.3331 28.7552 28.755 29.3334 27.9998 29.3334H3.99997C3.24477 29.3334 2.66664 28.7552 2.66664 28V10.6668C2.66664 9.91165 3.24477 9.33352 3.99997 9.33352ZM24.6665 17.3335C23.5623 17.3335 22.6665 18.2293 22.6665 19.3334C22.6665 20.4376 23.5623 21.3334 24.6665 21.3334C25.7706 21.3334 26.6664 20.4376 26.6664 19.3334C26.6664 18.2293 25.7706 17.3335 24.6665 17.3335Z"
                fill="#151515"
              />
            </svg>
          </div>
          <div>
            <p
              class="c5"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            </p>
            <div
              class="c6"
            >
              <p>
                hello world
              </p>
            </div>
          </div>
        </div>
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            <button
              aria-label="Add Funds"
              class="c9"
            >
              <span
                class="c10"
              >
                Add Funds
              </span>
            </button>
          </div>
          <div
            class="c8"
          >
            <button
              aria-label="Do it later"
              class="c11"
            >
              <span
                class="c10"
              >
                Do it later
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
